data/src/main/java/org/openmbee/sdvc/data/domains/global/Base.java

```diff
@@ -58,8 +59,53 @@ public void setModified(Instant modified) {
         this.modified = modified;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        try {
+            Field[] fields = getClass().getDeclaredFields();
+            for (Field field : fields) {
+                Object current = field.get(this);
+                Object value = field.get(o);
+                switch(field.getType().toString()) {
+                    case "String" :
+                    case "DecimalFormat" :
+                    case "BigDecimal":
+                        if ((current != null && !current.equals(value)) || (current == null && value != null)) {
+                            return false;
+                        }
+
```

Comment:
think a break is needed here, otherwise current.equals(value) can be true but it'll fall through to default and potentially return false

Reply:
Wouldn't we want it to return false if the loop goes to default and current != value?